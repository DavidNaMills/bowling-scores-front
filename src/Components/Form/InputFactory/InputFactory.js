import React from 'react';
import Input from '../elements/Input/Input';

const Inputfactory = ({ config, id, changed }) => {
    let element = null;

    switch (config.elementtype.toLowerCase()) {
        case 'input':
            element = <Input
                changed={changed}
                id={id}
                {...config.elementConfig}
                isValid={config.isValid}
                label={config.elementConfig.placeholder}
                value={config.value}
                error={config.hasErr}
                isRequired={config.shouldValidate}
            />
            break;
        default:
            element = <p>ERROR</p>
    }

    return (
        <React.Fragment>
            {element}
        </React.Fragment>
    )
};

export default Inputfactory;