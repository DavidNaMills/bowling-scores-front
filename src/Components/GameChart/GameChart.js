import React from 'react';
import {
    LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, Legend,
} from 'recharts';



const GameChart = ({players, data}) => {
    return (
        <div>
            <LineChart
                width={380}
                height={350}
                data={data.data}
                margin={{
                    top: 5, right: 5, left: 0, bottom: 0,
                }}
            >
                <CartesianGrid strokeDasharray="3 3" />
                <XAxis dataKey="game" width={40}/>
                <YAxis width={40}/>
                <Tooltip />
                {
                    data.players.map(x=>
                        <Line type="monotone" dataKey={players[x].name} stroke={`rgb(${players[x].color})`} fill={`rgb(${players[x].color})`} activeDot={{ r: 9 }} />
                    )
                }
            </LineChart>
        </div>
    )
};

export default GameChart;