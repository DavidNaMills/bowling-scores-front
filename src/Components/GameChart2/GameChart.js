import React from 'react';
import { Chart } from 'react-google-charts';

const GameChart = () => {

    return (
        <div>
            <Chart
                width={'100%'}
                height={'300px'}
                chartType='LineChart'
                loader={<div>Loading Chart</div>}
                data={[
                    ['x', 'David', 'Paul', 'Alan'],
                    [0, 0, 0, 34],
                    [1, 10, 5, 35],
                    [2, 23, 15, 24],
                    [3, 17, 9, 56],
                    [4, 50, 10, 80],
                    [5, null, null, 32],
                    [6, 23, 23, 3],
                    [7, 27, 19, 56],
                ]}
                options={{
                    legend: { position: 'bottom' },
                    chartArea:{left:10,top:10, bottom: 10,width:'95%',height:'100%'},
                    pointSize: 5,
                    series: {
                        0: { curveType: 'function', color: 'blue' },
                        1: { curveType: 'function', color: 'red' },
                        2: { curveType: 'function', color: 'green' },
                    },
                }}
            />
        </div>
    )
}

export default GameChart;