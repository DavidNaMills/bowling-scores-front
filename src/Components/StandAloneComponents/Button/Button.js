import React from 'react';
import classes from './Button.module.scss';



const Button = ({ click = () => { }, label, isDisabled = false, type = 'default' }) => {
    let style = [classes.button];

    if (isDisabled) {
        style = style.concat(classes.button__disabled);
    } else {

        switch (type.toLowerCase()) {
            case 'warning':
                style = style.concat(classes.button__warning);
                break;
            case 'danger':
                style = style.concat(classes.button__danger);
                break;
            default:
                style = style.concat(classes.button__default);
                break;
        }
    }

    return (
        <button onClick={click} disabled={isDisabled} className={style.join(' ')}>{label}</button>
    );
};

export default Button;