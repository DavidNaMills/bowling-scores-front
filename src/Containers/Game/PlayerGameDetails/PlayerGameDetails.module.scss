.playerDetails{
    &__width{
        width: 75vw;
        display: flex;
        flex-direction: column;
        margin: auto;
        padding-top: 20px;

        @media (max-width: 500px) {
            width: 100%;
        }
        
        @media (min-width: 1000px) {
            width: 80vw;
        }

        @media (min-width: 1200px) {
            width: 50vw;
        }
    }
}