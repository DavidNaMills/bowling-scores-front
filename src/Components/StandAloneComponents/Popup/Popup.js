import React from 'react';
import classes from './Popup.module.scss';

import useBodyLock from '../../../Hooks/useBodyLock/useBodyLock';
import Button from '../Button/Button';
import Title from '../Title/Title';

const Popup = ({ message, title = null, action1 = null, action2 = null, close = null }) => {
    useBodyLock();

    return (
        <div className={classes.popup__container}>
            <div className={classes.popup__module}>
                {close &&
                    <div className={classes.popup__closeBar}>
                        <p onClick={close} className={classes.popup__closeBar_cls}>CLOSE</p>
                    </div>
                }
                <div>

                    {title && <Title {...title} />}
                    <p className={classes.popup__message}>{message}</p>
                    <div className={classes.popup__btnContainer}>
                        {action1 && <Button {...action1} />}
                        {action2 && <Button {...action2} />}
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Popup;