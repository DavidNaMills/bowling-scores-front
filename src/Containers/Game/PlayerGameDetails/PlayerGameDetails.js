import React, { useMemo, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import * as actions from '../../../store/allActions';

import Title from '../../../Components/StandAloneComponents/Title/Title';
import Button from '../../../Components/StandAloneComponents/Button/Button';
import GameChart from '../../../Components/GameChart/GameChart';
import Table from '../../../Components/Table/Table';
import InputFactory from '../../../Components/Form/InputFactory/InputFactory';

import tableModifier from '../../../helpers/tableGamesDataModifier/tableGamesDataModifier';
import chartParser from '../../../helpers/chartDataModifier/chartDataModifier';
import singlePlayerScores from '../../../helpers/singlePlayerScores/singlePlayerScores';
import generateForm from '../../../helpers/generateAmendFormConfig/generateAmendFormConfig';
import objectToArray from '../../../helpers/objectToArray/objectToArray';


const tHeaders = ['Average', 'Pinfall']
const tHeaders2 = ['Score']


const PlayerGameDetails = (props) => {
    const dispatch = useDispatch();
    const removePlayerDispatch = (id) => dispatch(actions.removePlayer(id));
    const updateScoresDispatch = (data) => dispatch(actions.updateIndividualScore(data))

    const gameData = useSelector(state=>state.liveGame);
    const id = useMemo(() => props.location.state.id);
    const isLoading = false;
    const [showEdit, setShowEdit] = useState(false);

    const table1 = useMemo(() => ({
        headers: tHeaders,
        rows: tableModifier(gameData, id)
    }));

    const scoreData = useMemo(() => ({
        headers: tHeaders2,
        rows: singlePlayerScores(gameData, id)
    }));


    const updateScores = () => {
        /* {
            id:
            scores:{
                1: 45,
                3: 123
            }
        }*/
    }

    const removePlayerLocal = () => {
        props.history.push('/game');
        removePlayerDispatch(id);
    }


    const buildTable = () => (
        <Table
            data={scoreData}
            showRowNum
        />
    )
console.log(scoreData);
    const buildEditForm = () => (
        <div>
            <form onSubmit={updateScores}>
                {
                    objectToArray(generateForm(scoreData.rows)).map((x, i) =>
                        <InputFactory
                            key={i}
                            config={x.config}
                            id={x.id}
                            changed={updateScores}
                            type='number'
                            min={0}
                            max={300}
                        />, 0)
                }
                <Button label='Update' type='default' click={()=>{}} isFull isDisabled={isLoading} />
            </form>
            <Button label='Close' type='danger' click={() => setShowEdit(prev => !prev)} isFull isDisabled={isLoading} />
        </div>
    )

    return (
        <div>
            <Title label={gameData.players[id].name} ttlType='section' />
            <Table
                data={table1}
            />

            <GameChart
                players={gameData.players}
                data={chartParser(gameData, id)}
            />

            {
                showEdit
                    ? buildEditForm()
                    : buildTable()

            }
            {
                !showEdit &&
                <React.Fragment>

                    <br />
                    <Button isFull label={showEdit ? 'Close' : 'Edit scores'} type='warning' click={() => { setShowEdit(prev => !prev) }} />
                    <br />
                    <Button isFull label='Back' type='default' click={() => { props.history.goBack() }} />
                    <br />
                    <Button isFull label='Remove Player' type='danger' click={removePlayerLocal} />
                </React.Fragment>
            }
        </div>
    )
}

export default PlayerGameDetails;

/**
 *
 */
