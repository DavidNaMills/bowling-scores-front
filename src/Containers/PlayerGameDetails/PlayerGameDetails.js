import React, { useMemo, useState } from 'react';

import Title from '../../Components/StandAloneComponents/Title/Title';
import Button from '../../Components/StandAloneComponents/Button/Button';
import GameChart from '../../Components/GameChart/GameChart';
import Table from '../../Components/Table/Table';
import InputFactory from '../../Components/Form/InputFactory/InputFactory';

import tableModifier from '../../helpers/tableGamesDataModifier/tableGamesDataModifier';
import chartParser from '../../helpers/chartDataModifier/chartDataModifier';
import singlePlayerScores from '../../helpers/singlePlayerScores/singlePlayerScores';
import generateForm from '../../helpers/generateAmendFormConfig/generateAmendFormConfig';
import objectToArray from '../../helpers/objectToArray/objectToArray';

const tHeaders = ['Average', 'Pinfall']
const tHeaders2 = ['Score']


const PlayerGameDetails = (props) => {
    const id = useMemo(() => props.location.state.id);
    const gameData = useMemo(() => props.location.state.gameData);
    const isLoading = false;

    const [showEdit, setShowEdit] = useState(false);

    const table1 = useMemo(() => ({
        headers: tHeaders,
        rows: tableModifier(gameData, id)
    }));

    const scoreData = useMemo(() => ({
        headers: tHeaders2,
        rows: singlePlayerScores(gameData, id)
    }));


    const tempClick = () => {

    }


    const buildTable = () => (
        <Table
            data={scoreData}
            showRowNum
        />
    )

    const buildForm = () => (
        <div>
            <form onSubmit={tempClick}>
                {
                    objectToArray(generateForm(scoreData.rows)).map((x, i) =>
                        <InputFactory
                            key={i}
                            config={x.config}
                            id={x.id}
                            changed={tempClick}
                        />, 0)
                }
                <Button label='Update' type='default' click={tempClick} isFull isDisabled={isLoading} />
            </form>
            <Button label='Close' type='danger' click={() => setShowEdit(prev => !prev)} isFull isDisabled={isLoading} />
        </div>
    )

    return (
        <div>
            <Title label={gameData.players[id].name} ttlType='section' />
            <Table
                data={table1}
            />

            <GameChart
                players={gameData.players}
                data={chartParser(gameData, id)}
            />

            {
                showEdit
                    ? buildForm()
                    : buildTable()

            }
            {
                !showEdit &&
                <React.Fragment>

                    <br />
                    <Button isFull label={showEdit ? 'Close' : 'Edit scores'} type='warning' click={() => { setShowEdit(prev => !prev) }} />
                    <br />
                    <Button isFull label='Back' type='default' click={() => { props.history.goBack() }} />
                    <br />
                    <Button isFull label='Remove Player' type='danger' click={() => { alert('delete player') }} />
                </React.Fragment>
            }
        </div>
    )
}

export default PlayerGameDetails;

/**
 *
 */
