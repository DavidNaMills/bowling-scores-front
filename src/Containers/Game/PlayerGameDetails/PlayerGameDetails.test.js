import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

jest.mock('react-redux');
import {useSelector} from 'react-redux';

jest.mock('../../../Hooks/useShowHook/useShow');
import useShowHook from '../../../Hooks/useShowHook/useShow';

const original = window;
window.scrollTo = jest.fn();

import PlayerGameDetails from './PlayerGameDetails';
import Title from '../../../Components/StandAloneComponents/Title/Title';
import Button from '../../../Components/StandAloneComponents/Button/Button';
import Table from '../../../Components/Table/Table';
import GameChart from '../../../Components/GameChart/GameChart';
import InputFactory from '../../../Components/Form/InputFactory/InputFactory';
import Popup from '../../../Components/StandAloneComponents/Popup/Popup';


const mockLiveGame = {
    games: {
        1: { 'debb53c3-ad2a-4093-b89b-2a5025416f6c': { name: "gfdgfds", score: 12 } },
        2: { 'debb53c3-ad2a-4093-b89b-2a5025416f6c': { name: "gfdgfds", score: 24 } },
        3: { 'debb53c3-ad2a-4093-b89b-2a5025416f6c': { name: "gfdgfds", score: 154 } },
    },
    players: {
        'debb53c3-ad2a-4093-b89b-2a5025416f6c': { name: "gfdgfds", color: "17, 127, 14", id: "debb53c3-ad2a-4093-b89b-2a5025416f6c" }
    }
};

const mockUser = {
    token: '1d23saf1d23sa',
    user: {name: 'david', color: '132, 123, 123'}
};

const mockLocation = {state: {id: 'debb53c3-ad2a-4093-b89b-2a5025416f6c'}};


afterEach(()=>{
    jest.clearAllMocks();
    window = original;
});

describe('<playerGameDetails/> test suite', ()=>{
    beforeEach(()=>{
        useShowHook.mockImplementation(()=>({
            isShow: {
                showEdit: false,
                showPopup: false
            },
            changeShow: jest.fn()
        }))
    });

    it('matches snapshot', ()=>{
        useSelector.mockImplementation(()=>({
            liveGame: mockLiveGame,
            user: mockUser
        }));
        const comp = renderer.create(<PlayerGameDetails location={mockLocation}/>);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });
    
    it('exists', ()=>{
        const wrapper = mount(<PlayerGameDetails location={mockLocation}/>);
        expect(wrapper.exists).toBeTruthy();
    });
    
    describe('default view (isEdit===false)', ()=>{
        let wrapper;

        beforeEach(()=>{
            useSelector.mockImplementation(()=>({
                liveGame: mockLiveGame,
                user: mockUser
            }));
            wrapper = mount(<PlayerGameDetails location={mockLocation}/>);
        })
        
        it('renders a container <div> with class="contentContainer"', ()=>{
            const div = wrapper.find('div');
            expect(div.at(0).props().className).toEqual('contentContainer');
        });
        
        it('renders 1 buildBackButtons', ()=>{
            expect(wrapper.find('#buildDelete').length).toBe(1);
        });
        
        it('renders a <Title /> component', ()=>{
            expect(wrapper.find(Title).length).toBe(1);
        });

        it('renders 2 <Table/> components', ()=>{
            expect(wrapper.find(Table).length).toBe(2);
        });

        it('renders a <GameChart/> component', ()=>{
            expect(wrapper.find(GameChart).length).toBe(1);
        });

        it('Buttons have correct labels', ()=>{
            const btn = wrapper.find(Button);
            expect(btn.length).toBe(4);
            expect(btn.at(0).props().label).toEqual('Back');
            expect(btn.at(1).props().label).toEqual('Edit scores');
            expect(btn.at(2).props().label).toEqual('Back');
            expect(btn.at(3).props().label).toEqual('Remove Player');
        });
    });
    
    describe('showEdit is true', ()=>{
        let wrapper;

        beforeEach(()=>{
            useSelector.mockImplementation(()=>({
                liveGame: mockLiveGame,
                user: mockUser
            }));
            wrapper = mount(<PlayerGameDetails location={mockLocation}/>);
        })

        it('displays the form for updating. classname="form__container"', ()=>{
            useShowHook.mockImplementation(()=>({
                isShow: {
                    showEdit: true,
                    showPopup: false
                },
                changeShow: jest.fn()
            }));
            wrapper = mount(<PlayerGameDetails location={mockLocation}/>);
            expect(wrapper.find('.form__container').length).toBe(1);
        });
        
        it('renders a form with 3 <InputFactory/> components', ()=>{
            useShowHook.mockImplementation(()=>({
                isShow: {
                    showEdit: true,
                    showPopup: false
                },
                changeShow: jest.fn()
            }));
            wrapper = mount(<PlayerGameDetails location={mockLocation}/>);
            expect(wrapper.find('form').length).toBe(1);
            expect(wrapper.find(InputFactory).length).toBe(3);
        });

        it('renders 4 <Button/> components with correct labels', ()=>{
            useShowHook.mockImplementation(()=>({
                isShow: {
                    showEdit: true,
                    showPopup: false
                },
                changeShow: jest.fn()
            }));
                const btn = wrapper.find(Button);
                expect(btn.length).toBe(4);
                expect(btn.at(0).props().label).toEqual('Back');
                expect(btn.at(1).props().label).toEqual('Edit scores');
                expect(btn.at(2).props().label).toEqual('Back');
                expect(btn.at(3).props().label).toEqual('Remove Player');
        });
    });

    describe('popup display tests', ()=>{
        let wrapper;

        beforeEach(()=>{
            useSelector.mockImplementation(()=>({
                liveGame: mockLiveGame,
                user: mockUser
            }));
            wrapper = mount(<PlayerGameDetails location={mockLocation}/>);
        })

        it('displays the form for updating. classname="form__container"', ()=>{
            useShowHook.mockImplementation(()=>({
                isShow: {
                    showEdit: false,
                    showPopup: true
                },
                changeShow: jest.fn()
            }));
            wrapper = mount(<PlayerGameDetails location={mockLocation}/>);
            expect(wrapper.find(Popup).length).toBe(1);
        });
    });
});

// showPopup
