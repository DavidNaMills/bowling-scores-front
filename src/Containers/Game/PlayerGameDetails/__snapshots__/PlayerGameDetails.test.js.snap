// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<playerGameDetails/> test suite matches snapshot 1`] = `
<div
  className="contentContainer"
>
  <div
    className="smallExtra"
  >
    <button
      className="button button__isFull button__default"
      disabled={false}
      onClick={[Function]}
    >
      Back
    </button>
  </div>
  <div
    className="title"
  >
    <h2
      className="title__section"
    >
      gfdgfds
    </h2>
  </div>
  <div
    className="extra"
  >
    <div>
      
      <table
        className="table"
      >
        <thead>
          <tr>
            <td>
              Average
            </td>
            <td>
              Pinfall
            </td>
          </tr>
        </thead>
        <tbody>
          <tr
            className=""
            onClick={[Function]}
            style={
              Object {
                "background": "linear-gradient(180deg, rgba(255,255,255,1) 37%, rgba(17, 127, 14,1) 100%)",
              }
            }
          >
            <td>
              63
            </td>
            <td>
              190
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="extra centerDiv"
  >
    <div>
      <div
        className="recharts-wrapper"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "cursor": "default",
            "height": 350,
            "position": "relative",
            "width": 800,
          }
        }
      >
        <svg
          className="recharts-surface"
          height={350}
          version="1.1"
          viewBox="0 0 800 350"
          width={800}
        >
          <defs>
            <clipPath
              id="recharts2-clip"
            >
              <rect
                height={315}
                width={755}
                x={40}
                y={5}
              />
            </clipPath>
          </defs>
          <g
            className="recharts-cartesian-grid"
          >
            <g
              className="recharts-cartesian-grid-horizontal"
            >
              <line
                fill="none"
                height={315}
                stroke="#ccc"
                strokeDasharray="3 3"
                width={755}
                x={40}
                x1={40}
                x2={795}
                y={5}
                y1={320}
                y2={320}
              />
              <line
                fill="none"
                height={315}
                stroke="#ccc"
                strokeDasharray="3 3"
                width={755}
                x={40}
                x1={40}
                x2={795}
                y={5}
                y1={241.25}
                y2={241.25}
              />
              <line
                fill="none"
                height={315}
                stroke="#ccc"
                strokeDasharray="3 3"
                width={755}
                x={40}
                x1={40}
                x2={795}
                y={5}
                y1={162.5}
                y2={162.5}
              />
              <line
                fill="none"
                height={315}
                stroke="#ccc"
                strokeDasharray="3 3"
                width={755}
                x={40}
                x1={40}
                x2={795}
                y={5}
                y1={83.75}
                y2={83.75}
              />
              <line
                fill="none"
                height={315}
                stroke="#ccc"
                strokeDasharray="3 3"
                width={755}
                x={40}
                x1={40}
                x2={795}
                y={5}
                y1={5}
                y2={5}
              />
            </g>
            <g
              className="recharts-cartesian-grid-vertical"
            >
              <line
                fill="none"
                height={315}
                stroke="#ccc"
                strokeDasharray="3 3"
                width={755}
                x={40}
                x1={40}
                x2={40}
                y={5}
                y1={5}
                y2={320}
              />
              <line
                fill="none"
                height={315}
                stroke="#ccc"
                strokeDasharray="3 3"
                width={755}
                x={40}
                x1={417.5}
                x2={417.5}
                y={5}
                y1={5}
                y2={320}
              />
              <line
                fill="none"
                height={315}
                stroke="#ccc"
                strokeDasharray="3 3"
                width={755}
                x={40}
                x1={795}
                x2={795}
                y={5}
                y1={5}
                y2={320}
              />
            </g>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
          >
            <line
              className="recharts-cartesian-axis-line"
              fill="none"
              height={30}
              stroke="#666"
              width={755}
              x={40}
              x1={40}
              x2={795}
              y={320}
              y1={320}
              y2={320}
            />
            <g
              className="recharts-cartesian-axis-ticks"
            >
              <g
                className="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  className="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height={30}
                  stroke="#666"
                  width={755}
                  x={40}
                  x1={40}
                  x2={40}
                  y={320}
                  y1={326}
                  y2={320}
                />
                <text
                  className="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height={30}
                  stroke="none"
                  textAnchor="middle"
                  width={755}
                  x={40}
                  y={328}
                >
                  <tspan
                    dy="0.71em"
                    x={40}
                  >
                    1
                  </tspan>
                </text>
              </g>
              <g
                className="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  className="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height={30}
                  stroke="#666"
                  width={755}
                  x={40}
                  x1={417.5}
                  x2={417.5}
                  y={320}
                  y1={326}
                  y2={320}
                />
                <text
                  className="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height={30}
                  stroke="none"
                  textAnchor="middle"
                  width={755}
                  x={417.5}
                  y={328}
                >
                  <tspan
                    dy="0.71em"
                    x={417.5}
                  >
                    2
                  </tspan>
                </text>
              </g>
              <g
                className="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  className="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height={30}
                  stroke="#666"
                  width={755}
                  x={40}
                  x1={795}
                  x2={795}
                  y={320}
                  y1={326}
                  y2={320}
                />
                <text
                  className="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height={30}
                  stroke="none"
                  textAnchor="middle"
                  width={755}
                  x={795}
                  y={328}
                >
                  <tspan
                    dy="0.71em"
                    x={795}
                  >
                    3
                  </tspan>
                </text>
              </g>
            </g>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
          >
            <line
              className="recharts-cartesian-axis-line"
              fill="none"
              height={315}
              stroke="#666"
              width={40}
              x={0}
              x1={40}
              x2={40}
              y={5}
              y1={5}
              y2={320}
            />
            <g
              className="recharts-cartesian-axis-ticks"
            >
              <g
                className="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  className="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height={315}
                  stroke="#666"
                  width={40}
                  x={0}
                  x1={34}
                  x2={40}
                  y={5}
                  y1={320}
                  y2={320}
                />
                <text
                  className="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height={315}
                  stroke="none"
                  textAnchor="end"
                  width={40}
                  x={32}
                  y={320}
                >
                  <tspan
                    dy="0.355em"
                    x={32}
                  >
                    0
                  </tspan>
                </text>
              </g>
              <g
                className="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  className="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height={315}
                  stroke="#666"
                  width={40}
                  x={0}
                  x1={34}
                  x2={40}
                  y={5}
                  y1={241.25}
                  y2={241.25}
                />
                <text
                  className="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height={315}
                  stroke="none"
                  textAnchor="end"
                  width={40}
                  x={32}
                  y={241.25}
                >
                  <tspan
                    dy="0.355em"
                    x={32}
                  >
                    40
                  </tspan>
                </text>
              </g>
              <g
                className="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  className="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height={315}
                  stroke="#666"
                  width={40}
                  x={0}
                  x1={34}
                  x2={40}
                  y={5}
                  y1={162.5}
                  y2={162.5}
                />
                <text
                  className="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height={315}
                  stroke="none"
                  textAnchor="end"
                  width={40}
                  x={32}
                  y={162.5}
                >
                  <tspan
                    dy="0.355em"
                    x={32}
                  >
                    80
                  </tspan>
                </text>
              </g>
              <g
                className="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  className="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height={315}
                  stroke="#666"
                  width={40}
                  x={0}
                  x1={34}
                  x2={40}
                  y={5}
                  y1={83.75}
                  y2={83.75}
                />
                <text
                  className="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height={315}
                  stroke="none"
                  textAnchor="end"
                  width={40}
                  x={32}
                  y={83.75}
                >
                  <tspan
                    dy="0.355em"
                    x={32}
                  >
                    120
                  </tspan>
                </text>
              </g>
              <g
                className="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  className="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height={315}
                  stroke="#666"
                  width={40}
                  x={0}
                  x1={34}
                  x2={40}
                  y={5}
                  y1={5}
                  y2={5}
                />
                <text
                  className="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height={315}
                  stroke="none"
                  textAnchor="end"
                  width={40}
                  x={32}
                  y={5}
                >
                  <tspan
                    dy="0.355em"
                    x={32}
                  >
                    160
                  </tspan>
                </text>
              </g>
            </g>
          </g>
          <g
            className="recharts-layer recharts-line"
          >
            <path
              className="recharts-curve recharts-line-curve"
              clipPath={null}
              d="M40,296.375C165.83333333333331,292.4375,291.6666666666667,288.5,417.5,272.75C543.3333333333334,257,669.1666666666666,136.90625,795,16.8125"
              fill="none"
              height={315}
              stroke="rgb(17, 127, 14)"
              strokeWidth={1}
              width={755}
            />
          </g>
        </svg>
        <div
          className="recharts-tooltip-wrapper"
          style={
            Object {
              "MozTransform": "translate(undefinedpx, undefinedpx)",
              "OTransform": "translate(undefinedpx, undefinedpx)",
              "WebkitTransform": "translate(undefinedpx, undefinedpx)",
              "msTransform": "translate(undefinedpx, undefinedpx)",
              "pointerEvents": "none",
              "position": "absolute",
              "top": 0,
              "transform": "translate(undefinedpx, undefinedpx)",
              "visibility": "hidden",
            }
          }
        >
          <div
            className="recharts-default-tooltip"
            style={
              Object {
                "backgroundColor": "#fff",
                "border": "1px solid #ccc",
                "margin": 0,
                "padding": 10,
                "whiteSpace": "nowrap",
              }
            }
          >
            <p
              className="recharts-tooltip-label"
              style={
                Object {
                  "margin": 0,
                }
              }
            >
              
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="gameDetails__extra"
  >
    <div>
      
      <table
        className="table"
      >
        <thead>
          <tr>
            <td />
            <td>
              Score
            </td>
          </tr>
        </thead>
        <tbody>
          <tr
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            <td>
              1
            </td>
            <td>
              12
            </td>
          </tr>
          <tr
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            <td>
              2
            </td>
            <td>
              24
            </td>
          </tr>
          <tr
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            <td>
              3
            </td>
            <td>
              154
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="extra"
  >
    <button
      className="button button__isFull button__warning"
      disabled={false}
      onClick={[Function]}
    >
      Edit scores
    </button>
  </div>
  <div
    className="smallExtra"
  >
    <button
      className="button button__isFull button__default"
      disabled={false}
      onClick={[Function]}
    >
      Back
    </button>
  </div>
  <div
    className="superLarge"
    id="buildDelete"
  >
    <button
      className="button button__isFull button__danger"
      disabled={false}
      onClick={[Function]}
    >
      Remove Player
    </button>
  </div>
</div>
`;
